wordpress-mobile/WordPress-iOS
# Adapt WebViewAuthenticator for atomic

Fixes #13566 

This PR implements new code to request special auth cookies for Atomic Private Sites.

This PR also modularizes and simplifies the WebViewAuthenticator code, so that it doesn't need to know the details of how to request authorization cookies.

We're no longer requesting the cookies directly through the web view but instead using ephemeral URL Sessions within services.  This is good in theory because it allows our code to request those authorization cookies regardless of if we are using a web view.

## To test:

### Test 1:

1. Create an atomic private site.
2. Make sure it's private.
3. Preview the site, a page and a post.

Make sure you can preview the content just fine.
Do the same test for post and page previews.

### Test 2:

Same but with a WP.com site.
Do the same test for post and page previews.

### Test 3:

Same but with a self-hosted site connected through XMLRPC.
Do the same test for post and page previews.

### Test 4:

Same but with a self-hosted site connected through Jetpack.
Do the same test for post and page previews.

### Test 5:

1. Create a self hosted site.
2. Connect in the app through XMLRPC.
3. Go to the site's dashboard in the app and tap "Stats".
4. Complete the Jetpack installation and connection.

Make sure it works fine.

### Test 6:

Follow a private site.
Open posts from it in the reader.

## PR submission checklist:
